### ISSUE TITLE
Package request: WeChat

### ISSUE THREAD

**Issue by Seeker0472** [NixOS/nixpkgs#354234]
**Project description**
WeChat (or Weixin/微信 in Chinese)  is a Chinese [instant messaging](https://en.wikipedia.org/wiki/Instant_messaging), [social media](https://en.wikipedia.org/wiki/Social_media), and [mobile payment](https://en.wikipedia.org/wiki/Mobile_payment) [app](https://en.wikipedia.org/wiki/Mobile_app) developed by [Tencent](https://en.wikipedia.org/wiki/Tencent) and is widely used in China.
And Tencent has released the official linux test version of Wechat recently(Which is different from the wechat-uos from nixpkgs).

**Metadata**

* homepage URL: https://linux.weixin.qq.com/en
* license: unknown
* platforms: linux

**Comment by jian-lin** [comment]
cc wechat-uos maintainers @pokon548 @xddxdd

**Comment by UlyssesZh** [comment]
```nix
  inherit (stdenv.hostPlatform) system;

  src = fetchurl {
    url = {
      x86_64-linux = "https://web.archive.org/web/20241107191621/https://dldir1v6.qq.com/weixin/Universal/Linux/WeChatLinux_x86_64.AppImage";
      aarch64-linux = "https://web.archive.org/web/20241107192345/https://dldir1v6.qq.com/weixin/Universal/Linux/WeChatLinux_arm64.AppImage";
      x86_64-darwin = "https://web.archive.org/web/20241107193311/https://dldir1v6.qq.com/weixin/Universal/Mac/WeChatMac_x86_64.dmg";
      aarch64-darwin = "http://web.archive.org/web/20241107194014/https://dldir1v6.qq.com/weixin/Universal/Mac/WeChatMac_arm64.dmg";
    }.${system};
    # ...
  }
```
@pluiedev

**Comment by pluiedev** [comment]
Ah yes, the classic trick known as "just use the Wayback Machine"... Fair enough, I might take a look at it

**Comment by UlyssesZh** [comment]
Yeah I think this has been the standard way of handling this in nixpkgs. Those links are created by me today, should be new enough.

**Comment by pokon548** [comment]
> @aucub* However, I don't believe that's the case. The wechat-uos package in the master branch of nixpkgs was downloaded from [uniontech.com](https://uniontech.com/)* and was originally designed exclusively for UOS, requiring a UOS license to operate.

Actually `4.0.0.21` does not require `libuosdevicea` and `license.tar.gz` anymore. I just forget to remove them and `wechat-uos` works normally. Will upstream changes it in next PR.

> Looks like download links are not versioned in any way — doesn't sound stable enough for inclusion in Nixpkgs
I had reached officials for this problem, but had not received any response yet.

**Comment by darkclip** [comment]
tried the official version, more functions are available, seems no bwrap anymore, overall i think it's better.
